src/test/java/usecasetesting/CloseUseCaseTest.java

```diff
@@ -0,0 +1,137 @@
+
+public class CloseUseCaseTest {
+    final static int CLIENT_ID = 21345678;
+    final static int ATTORNEY_ID = 11345678;
+    final static int SECOND_ATTORNEY_ID = 12222222;
+    final static int QUESTION_ID = 323456789;
+    final static int CLOSED_QUESTION_ID = 333333333;
+    private QuestionGateway questionGateway;
+    private PostGateway postGateway;
+    private PostFactory postFactory;
+    private UserGatewayFactory userGatewayFactory;
+    private ClientGateway clientGateway;
+    private AttorneyGateway attorneyGateway;
+    private HomePageOutputBoundary homePageOutputBoundary;
+    private CloseInputBoundary closeInputBoundary;
+
+    public void setUpCloseUseCase(){
+
+        questionGateway = new QuestionRepo();
+        postGateway = new PostRepo();
+        postFactory = new PostFactory();
+        userGatewayFactory = new UserGatewayFactory();
+        clientGateway = new ClientRepository();
+        attorneyGateway = new AttorneyRepository();
+        clientGateway.deleteAllUser();
+        questionGateway.deleteAllQuestion();
+        postGateway.deleteAllPost();
+        attorneyGateway.deleteAllUser();
+        homePageOutputBoundary = new HomePageOutputBoundary() {
+            @Override
+            public void setControlContainer(ControlContainer controlContainer) {
+
+            }
+
+            @Override
+            public HomePageResponseModel prepareFail(String msg) {
+                assertEquals("You cannot close this question!", msg);
+                return null;
+            }
+
+            @Override
+            public HomePageResponseModel prepareSuccess(HomePageResponseModel homePageResponseModel) {
+                return null;
+            }
+        };
+        closeInputBoundary = new CloseQuestionInteractor(questionGateway, homePageOutputBoundary, userGatewayFactory);
+
+        Question question = new Question();
+        question.setQuestionId(QUESTION_ID);
+
+
+        Question closedQuestion = new Question();
+        closedQuestion.setQuestionId(CLOSED_QUESTION_ID);
+        questionGateway.saveQuestion(closedQuestion);
+
+        Client client = new Client();
+        client.setUserId(CLIENT_ID);
+        client.addQuestion(question);
+        clientGateway.addUser(client);
+
+        Attorney attorney = new Attorney();
+        attorney.setUserId(ATTORNEY_ID);
+        attorneyGateway.addUser(attorney);
+
+        Attorney secondAttorney = new Attorney();
+        attorney.setUserId(SECOND_ATTORNEY_ID);
+        attorneyGateway.addUser(secondAttorney);
+
+        question.setTakenByAttorney(ATTORNEY_ID);
+        question.setTaken(true);
+        questionGateway.saveQuestion(question);
+    }
+    @Test
+    public void TestClientCloseableQuestion(){
+        setUpCloseUseCase();
+
+        CloseRequestModel inputData = new CloseRequestModel(QUESTION_ID, CLIENT_ID);
+
+        closeInputBoundary.closeQuestion(inputData);
+        Question question = questionGateway.getQuestion(QUESTION_ID);
```

Comment:
I recommend including tests can verify the changes to the database, as that would help catch potential errors in the use case classes. This is highly important to the overall functionality of the code as a result of the nature of our program. Being able to catch cases where certain attributes of entities fail to be updated is extremely important.